packages/vector_graphics/lib/src/render_vector_graphics.dart

```diff
@@ -112,26 +134,34 @@ class RenderVectorGraphic extends RenderBox {
 
   Future<void>? _pendingRasterUpdate;
 
-  // Re-create the raster for a given SVG if the target size
+  // Re-create the raster for a given vector graphic if the target size
   // is sufficiently different.
   Future<void> _maybeUpdateRaster(Size desiredSize) async {
-    double scale = 1.0;
-    if (desiredSize != pictureInfo.size) {
-      scale = math.min(
-        desiredSize.width / pictureInfo.size.width,
-        desiredSize.height / pictureInfo.size.height,
-      );
-    }
-    final int scaledWidth = (pictureInfo.size.width * scale).round();
-    final int scaledHeight = (pictureInfo.size.height * scale).round();
+    final int scaledWidth =
+        (pictureInfo.size.width * devicePixelRatio / scale).round();
+    final int scaledHeight =
+        (pictureInfo.size.height * devicePixelRatio / scale).round();
     if (_lastRasterizedSize != null &&
         _lastRasterizedSize!.width == scaledWidth &&
         _lastRasterizedSize!.height == scaledHeight) {
       return;
     }
-    final ui.Image result = await pictureInfo.picture.toImage(
-        (scaledWidth * devicePixelRatio).round(),
-        (scaledHeight * devicePixelRatio).round());
+    // In order to scale a picture, it must be placed in a new picture
+    // with a transform applied. Surprisingly, the height and width
+    // arguments of Picture.toImage do not control the resolution that the
+    // picture is rendered at, instead it controls how much of the picture to
+    // capture in a raster.
+    final ui.PictureRecorder recorder = ui.PictureRecorder();
+    final ui.Canvas canvas = ui.Canvas(recorder);
+
+    final double totalScale = devicePixelRatio / scale;
+    canvas.transform(
+        Matrix4.diagonal3Values(totalScale, totalScale, totalScale).storage);
```

Comment:
haha I got so distracted by the matrix multiply that I forgot you could also just use `canvas.scale` but it's ok.